### DISCUSSION TITLE
where can I find my bookmarks in v3?

### DISCUSSION THREAD

**Discussion by bluestarmaki** [bluecaret/carettab#347] [Help and Support] [answered]
hi my caret tab updated to v4 yesterday. It went stuck, but now it works.
But I can't see any bookmarks as it was in v3. I wonder if they are all gone. v4 seems blank to me. Should I set the bookmarks from zero? :-( Or where can I find them? and load them into v 4? Thank you.
![image](https://github.com/bluecaret/carettab/assets/140778646/d631bb61-7955-4dcf-8fde-de8b7be6ab9a)

**Comment by bluecaret** [comment, marked as answer]
Hello, unfortunately this update required a reset due to various complexities in the change to how widgets are setup now. The update **should** have attempted to migrate your bookmarks and leave them in a disabled widget in the settings dashboard that  you could then re-enable. However it seems this has not worked for everyone and I apologize for that. You may have to readd the widgets from scratch again. If you open the settings panel from the bottom right, and add the Quick Link option, you will see an option to choose the type, where you can choose between Quick Links and Bookmarks Bar (where it pulls from the bookmarks saved in the browser). I hope that helps. If you need further assistance please let me know.

**Comment by jimenaffpp** [comment]
This happened already once in the past. So, bye bye Care Tab

**Reply by bluecaret** [reply]
Yes I'm sorry about that, this was required to allow better enhancement for widgets for everyone. Past decisions were making it difficult to offer the customization I wanted to provide so difficult decisions now had to be made. This extension was started when I was a new developer and I've learned a lot since then, if I wanted to continue to WANT to work on this as a project, I needed to make the updates. 

Please don't jump on other threads to make unnecessary goodbye comments. What's done is done, I can't redo it. I didn't spend months of hard work on this update with the goal of screwing everyone over.
